### PR TITLE
KAFKA-4700: Don't drop security configs in `StreamsKafkaClient`

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @guozhangwang.

**Comment by enothereska** [comment]
2 things: can we add a 1 sentence description to the JIRA? And is it common to test clients that support security with an unit/integration test? @ijuma you don't need to write the test, just wondering what best practices for testing this are. Thanks.

**Comment by ijuma** [comment]
@enothereska, I updated the JIRA (with more than one sentence :)). There should be integration tests for Streams and security yes. @guozhangwang said that there's a JIRA for it. In the absence of those, I wrote a unit test that verifies that the security configs are now being passed (it's easy to modify the code to verify that they were not being passed before).

**Comment by ijuma** [comment]
By the way, a future improvement is to allow the admin client configs to be namespaced like the producer and consumer configs can be namespaced currently.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClientTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+public class StreamsKafkaClientTest {
+
+    @Test
+    public void testConfigFromStreamsConfig() {
+        Properties props = new Properties();
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClientTest.java]: I know this is a bit picky, but we (mostly @mjsax and i) are trying to encourage everyone to use `final` where possible

**Review by dguy — COMMENTED** [review]
One very picky comment left.
Also, i'd add one more test where you configure it with a different `SaslConfigs.SASL_MECHANISM` - this is to ensure that "PLAIN" isn't hard-coded

**Comment by enothereska** [comment]
LGTM

**Comment by ijuma** [comment]
Thanks for the reviews, addressed your feedback @dguy.

**Review by dguy — APPROVED** [review]
LGTM

**Comment by ijuma** [comment]
retest this please

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
LGTM! Merged to trunk and 0.10.2.

**Pull request closed without merging**